crates/bevy_ui/src/focus.rs

```diff
@@ -111,7 +123,21 @@ pub fn ui_focus_system(
     let mut moused_over_z_sorted_nodes = node_query
         .iter_mut()
         .filter_map(
-            |(entity, node, global_transform, interaction, focus_policy, clip)| {
+            |(entity, node, global_transform, interaction, focus_policy, clip, visibility)| {
+                // Nodes that are not rendered should not be interactable
+                if let Some(computed_visibility) = visibility {
+                    if !computed_visibility.is_visible() {
+                        // Reset their interaction to None to avoid strange stuck state
+                        if let Some(mut interaction) = interaction {
+                            if *interaction != Interaction::None {
+                                *interaction = Interaction::None;
+                            }
+                        }
+
+                        return None;
```

Comment:
This would be simpler and skipping the nested branching to just assign `Interaction::None`, as it is small data-wise, should likely be faster:
```suggestion
                        *interaction = Interaction::None;
                        return None;
```

Reply:
I'm not sure this has the right change detection behavior: that's why this code is so convoluted. I don't want this triggering `Interaction` change detection every update.